src/MeshlessFV/MfvMusclSimulation.cpp

```diff
@@ -102,6 +102,15 @@ void MfvMusclSimulation<ndim>::MainLoop(void)
   // Advance N-body particle positions
   nbody->AdvanceParticles(n, nbody->Nnbody, t, timestep, nbody->nbodydata);
 
+  // Reset rebuild tree flag in preparation for next timestep
+  rebuild_tree        = false;
```

Comment:
Because this is the end of the timestep, and it is just before we inject the new particles. The flag needs to be reset at a certain point (see SPH main loop), and this is the most logical place